spec/unit/puppet/provider/powershell_input_spec.rb

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env ruby
+require 'spec_helper'
+require 'puppet/type'
+require 'puppet/type/base_dsc'
+
+example_dsc_resource =Puppet::Type.type(:dsc_file)
+
+describe 'powershell_provider' do
+
+  let :dsc_file do
+    Puppet::Type.type(:dsc_file).new(
+      :name => 'fio',
+      :dsc_ensure          => 'Present',
+      :dsc_destinationpath => 'C:/test.file',
+    )
+  end
+
+  describe "powershell provider" do
+
+    before(:each) do
+      @provider = example_dsc_resource.provider(:powershell).new(dsc_file)
+    end
+
+    describe "when quotes are present" do
+
+      it "should handle single quotes" do
+        dsc_file[:dsc_contents] = "The 'Cats' go 'meow'!"
+        expect(@provider.ps_script_content('test')).to match(/contents = 'The ''Cats'' go ''meow''!'/)
```

Comment:
I'd prefer this call be to `@provider.format_dsc_value` - possibly with an expect on `dsc_value.class.name` that returns `String`.